### PR TITLE
Merged SideBarLayoutContext and PaneMenulItem

### PR DESCRIPTION
**Pull request by severinoAmmirati**
- Removed `SideBarLayoutContext` as can be replaced by `PanelSideBarLayoutContext`
- Merged `PaneMenulItem` into `PanelItem`

### CODE REVIEW

**Comment by manni497** [comment]
@severinoAmmirati As discussed, please merge also PanelSideBarContext and PanelSideBarMenuContext :D

**Review by manni497 — CHANGES_REQUESTED** [review]

File: src/lib/Layout/PanelSideBarLayout/PanelSideBar/Definitions/PanelSideBarMenuItem.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { IconProp } from "@fortawesome/fontawesome-svg-core";
-import { ReactNode } from "react";
+import {ReactNode} from "react";
 
 export type PanelItem<TPanelItem = Record<string, unknown>, TMenuItem = Record<string, unknown>> = TPanelItem & {
   /**
    * The panel icon.
    */
-  icon: IconProp;
+  icon?: IconProp;
```
[the remark this answers is not in the document]
Comment (severinoAmmirati) [reply src/lib/Layout/PanelSideBarLayout/PanelSideBar/Definitions/PanelSideBarMenuItem.ts:8]: Icons are optional in both left and right sidebar, and seems that no exception are thrown
image

[the remark this answers is not in the document]
Comment (severinoAmmirati) [reply src/lib/Layout/PanelSideBarLayout/PanelSideBar/Definitions/PanelSideBarMenuItem.ts:8]: @manni497 Exception added

File: src/lib/Layout/PanelSideBarLayout/PanelSideBar/PanelSideBarItem.tsx [hunk 2]
```diff
@@ -1,50 +1,49 @@
...
-  onClick?: (menuItem: PanelMenuItem<unknown>) => void;
+  onClick?: (menuItem: PanelItem<unknown>) => void;
   depth?: number;
   active?: boolean;
   toggledItemIds: string[];
 }
 
 const PanelSideBarItem = (props: PanelSideBarItemProps) => {
-  const { depth = 0, item, LinkRenderer, onClick, toggledItemIds = [] } = props;
+  const { depth = 0, children, LinkRenderer, onClick, toggledItemIds = [] } = props;
```
[the remark this answers is not in the document]
Comment (severinoAmmirati) [reply src/lib/Layout/PanelSideBarLayout/PanelSideBar/PanelSideBarItem.tsx]: @manni497 Renamed as item

File: src/lib/Layout/PanelSideBarLayout/PanelSideBar/PanelSideBarItem.tsx [hunk 3]
```diff
@@ -1,50 +1,49 @@
...
             </a>
           </div>
         ) : (
           <>
-            <LinkRenderer item={item}>
+            <LinkRenderer item={children}>
               <span className="nav-link">
-                {item.icon && <FontAwesomeIcon icon={item.icon as IconProp} className="me-2" />}
-                {item.title}
+                {children.icon && <FontAwesomeIcon icon={children.icon as IconProp} className="me-2" />}
```
[the remark this answers is not in the document]
Comment (severinoAmmirati) [reply src/lib/Layout/PanelSideBarLayout/PanelSideBar/PanelSideBarItem.tsx]: @manni497 Not needed, removed

File: src/lib/Layout/PanelSideBarLayout/PanelSideBar/Context/PanelSideBarContext.tsx [hunk 4]
```diff
@@ -21,28 +21,52 @@ export interface PanelSideBarContextProps<TPanelItem, TMenuItem> {
...
    * The children elements.
    */
   children: React.ReactNode;
 }
 
 export const PanelSideBarProvider = <TPanelItem, TMenuItem>(props: PanelSideBarMenuProviderProps<TPanelItem, TMenuItem>) => {
-  const { children, globalItems, LinkRenderer } = props;
+  const { children, globalItems, LinkRenderer, brand = null, footer = null, userDropDownMenu, userDropDownMenuToggle, topBarCustomItems } = props;
+
+  //const firstActivePanel = globalItems.find((x) => x.children?.find((y) => y.active)) ?? globalItems.find((x) => x.id);
```
[the remark this answers is not in the document]
Comment (severinoAmmirati) [reply src/lib/Layout/PanelSideBarLayout/PanelSideBar/Context/PanelSideBarContext.tsx]: @manni497 Logic improved, let me know if it's good for you

**Review by manni497 — APPROVED** [review]

**Review by neotob — APPROVED** [review]

**Pull request merged**